### PR TITLE
[FLINK-28465] Wait compaction finished in endInput

### PR DESCRIPTION
**Pull request by LadyForest**
Under batch mode, the framework should always synchronize the compaction result before committing, and it is annoying to let users set this value to true.

### CODE REVIEW

**Review by JingsongLi — COMMENTED** [review]
I think we can add a `prepareCommit(boolean endInput)` to RecordWriter.
When end input, we can wait compaction.

File: flink-table-store-connector/src/main/java/org/apache/flink/table/store/connector/AbstractTableStoreFactory.java [hunk 1]
```diff
@@ -132,9 +133,14 @@ private static void validateFileStoreContinuous(Configuration options) {
     }
 
     static FileStoreTable buildFileStoreTable(DynamicTableFactory.Context context) {
-        FileStoreTable table =
-                FileStoreTableFactory.create(
-                        Configuration.fromMap(context.getCatalogTable().getOptions()));
+        boolean batch =
+                context.getConfiguration().get(ExecutionOptions.RUNTIME_MODE)
+                        == RuntimeExecutionMode.BATCH;
+        Map<String, String> options = new HashMap<>(context.getCatalogTable().getOptions());
```
Comment (SteNicholas) [inline flink-table-store-connector/src/main/java/org/apache/flink/table/store/connector/AbstractTableStoreFactory.java:139]:
```suggestion
        Configuration configuration = Configuration.fromMap(context.getCatalogTable().getOptions());
        configuration.setBoolean(CoreOptions.COMMIT_FORCE_COMPACT, batch);
        FileStoreTable table = FileStoreTableFactory.create(configuration);
```

File: flink-table-store-connector/src/main/java/org/apache/flink/table/store/connector/AbstractTableStoreFactory.java [hunk 2]
```diff
@@ -132,9 +133,14 @@ private static void validateFileStoreContinuous(Configuration options) {
...
+        if (batch) {
+            options.put(CoreOptions.COMMIT_FORCE_COMPACT.key(), String.valueOf(true));
```
Comment (SteNicholas) [inline flink-table-store-connector/src/main/java/org/apache/flink/table/store/connector/AbstractTableStoreFactory.java:141]: Does the config option `COMMIT_FORCE_COMPACT` need to explain in the description that the default value is true in BATCH mode?

**Review by JingsongLi — APPROVED** [review]
Looks good to me!

**Pull request merged**